src/modules/facility/dto/create-facility-request.dto.ts

```diff
@@ -0,0 +1,190 @@
+import { DateOnlyString } from '@ukef/helpers';
+
+export type CreateFacilityRequest = CreateFacilityRequestItem[];
+
+// TODO APIM-69: description improvements if possible
+export class CreateFacilityRequestItem {
+  @ValidatedDealIdentifierApiProperty({
+    description: 'The identifier of the deal for the facility.',
+  })
+  dealIdentifier: string;
+
+  @ValidatedFacilityIdentifierApiProperty({
+    description: 'The identifier of the facility to create.',
+  })
+  facilityIdentifier: string;
+
+  @ValidatedStringApiProperty({
+    description: 'The identifier of the portfolio that owns the facility.',
+    example: PROPERTIES.GLOBAL.portfolioIdentifier,
+    required: false,
+    default: PROPERTIES.GLOBAL.portfolioIdentifier,
+    maxLength: 2,
+  })
+  portfolioIdentifier?: string; // TODO APIM-69: is it right to allow the user to specify a different portfolio id?
+
+  @ValidatedPartyIdentifierApiProperty({
+    description:
+      'Look up the Obligors ACBS Customer record using Party URN on the Deal = Alternate Customer Id (J$MSZI). Use ACBS Customer Id (J$MRUI) Note there may be multiple customers in ACBS with the same Party URN. Use the first record found.',
+  })
+  dealBorrowerIdentifier: string;
+
+  @ValidatedStringApiProperty({
+    description: `The facility type, e.g. '250' for BOND.`,
+    example: '250',
+    maxLength: 3,
+  })
+  productTypeId: string;
+
+  @ValidatedStringApiProperty({
+    description: 'The name or description of the facility type.',
+    example: 'BOND',
+    maxLength: 13,
+  })
+  productTypeName: string;
+
+  @ValidatedStringApiProperty({
+    description: 'Credit Period',
+    example: '1',
+    maxLength: 12,
+  })
+  exposurePeriod: string;
+
+  @ValidatedCurrencyApiProperty({
+    description: 'Facility Currency Code e.g. CAD, USD.',
+  })
+  currency: string;
+
+  @ValidatedStringApiProperty({
+    description: 'The industry classification code of the obligor.',
+    example: '0116',
+    maxLength: 10,
+  })
+  obligorIndustryClassification: string;
+
+  @ValidatedDateOnlyApiProperty({
+    description: `The date from which the borrower can draw funds from the facility. If this is a future date then it will be replaced with today's date.`,
+  })
+  effectiveDate: DateOnlyString;
+
+  @ValidatedDateOnlyApiProperty({
+    description: 'The expiration date of the facility.',
+  })
+  guaranteeExpiryDate: DateOnlyString;
+
+  @ValidatedDateOnlyApiProperty({
+    description: 'The end date of the next quarter.',
+  })
+  nextQuarterEndDate: DateOnlyString;
+
+  @ValidatedNumberApiProperty({
+    description: 'The overall limit for the facility.',
+    minimum: 0,
+  })
+  maximumLiability: number;
+
+  @ValidatedPartyIdentifierApiProperty({
+    // TODO APIM-69: is this definitely a party id?
+    description: 'The party identifier of the Agent Bank for this facility.',
+  })
+  agentBankIdentifier: string;
```

Comment:
@avaitonis do you know if this should be a party identifier (and if we should validate it as if it is a party identifier)?

Reply:
ACBS field is called agentBankPartyIdentifier so it looks like ACBS party id.
Agent is available just for some deals. It might be empty in most cases.
Default value 00000000 means first party record UKEF (aka ECGD). Not sure if default value is used in current code.

Reply:
Looks like agentBankIdentifier is required field, so we don't need default value for it.

About validation, we have quite a lot of ACBS party fields, maybe validation is not worth the effort.

Reply:
Thanks Audrius - I've changed the validation to match Mulesoft's validation then (just that max # of characters is 10). I kept the validation on `obligorPartyIdentifier` to be _length exactly 8_ though, as I believe this should be correct

Reply:
It's fine, in future keeping 8 characters for all party ids might be nice consistency.